crates/bevy_ecs/src/component.rs

```diff
@@ -2070,42 +2086,65 @@ pub fn component_clone_via_clone<C: Clone + Component>(
 ///
 /// See [`ComponentCloneHandlers`] for more details.
 #[cfg(feature = "bevy_reflect")]
-pub fn component_clone_via_reflect(world: &mut DeferredWorld, entity_cloner: &EntityCloner) {
-    let component_id = entity_cloner.component_id();
-    let source = entity_cloner.source();
-    let target = entity_cloner.target();
-    world.commands().queue(move |world: &mut World| {
-        world.resource_scope::<crate::reflect::AppTypeRegistry, ()>(|world, registry| {
+pub fn component_clone_via_reflect(world: &mut DeferredWorld, ctx: &mut ComponentCloneCtx) {
+    fn clone_fast(ctx: &mut ComponentCloneCtx) -> Option<()> {
```

Comment:
I'd like some comments here about why this might fail to work, causing us to fallback to the slow mechanism.